app/commands/update_last_month_datasource_expenses.py

```diff
@@ -0,0 +1,139 @@
+
+    for organization in organizations:
+        if organization.linked_organization_id is None:
+            logger.warning(
+                "Organization %s has no linked organization ID. Skipping...", organization.id
+            )
+            continue
+
+        try:
+            logger.info("Fetching datasources for organization %s", organization.id)
+            response = await optscale_client.fetch_datasources_for_organization(
+                organization.linked_organization_id
+            )
+        except HTTPStatusError as exc:
+            response = exc.response
+
+            if exc.response.status_code == status.HTTP_404_NOT_FOUND:
+                logger.warning(
+                    "Organization %s not found on Optscale. Skipping...", organization.id
+                )
+            else:
+                logger.exception(
+                    "Unexpected error occurred fetching datasources for organization %s",
+                    organization.id,
+                )
+
+            continue
+
+        response_datasources = response.json()["cloud_accounts"]
+        logger.info(
+            "Fetched %d datasources for organization %s",
+            len(response_datasources),
+            organization.id,
+        )
+
+        datasources_per_organization_id[organization.id] = response_datasources
+
+    return datasources_per_organization_id
+
+
+async def store_datasource_expenses(
+    datasource_expense_handler: DatasourceExpenseHandler,
+    datasources_per_organization_id: dict[str, list[dict]],
+    year: int,
+    month: int,
+) -> None:
+    expenses_to_upsert = [
+        {
+            "id": DatasourceExpense.generate_human_readable_pk(),
+            "datasource_id": datasource["id"],
+            "organization_id": organization_id,
+            "year": year,
+            "month": month,
+            "month_expenses": datasource["details"]["cost"],
+            "created_at": datetime.now(UTC),
+            "updated_at": datetime.now(UTC),
+        }
+        for organization_id, datasources in datasources_per_organization_id.items()
+        for datasource in datasources
+    ]
+
+    logger.info("Prepared %d expenses to upsert", len(expenses_to_upsert))
+    await datasource_expense_handler.bulk_upsert(expenses_to_upsert)
+
+
+async def main(db_engine: AsyncEngine, settings: Settings) -> None:
+    sessionmaker = get_sessionmaker(db_engine)
+    today = datetime.now(UTC).date()
+
+    async with sessionmaker() as session:
+        datasource_expense_handler = DatasourceExpenseHandler(session)
+        organization_handler = OrganizationHandler(session)
+
+        async with session.begin():
+            logger.info("Querying organizations with no recent datasource expenses")
+            organizations = await organization_handler.query_db(
+                where_clauses=[
+                    ~Organization.datasource_expenses.any(DatasourceExpense.updated_at >= today)
```

Comment:
If I understood, do you want to get only organization where at least one datasource was not updated today right ? in that case, is it a bit clearer to express the query like 
`Organization.datasource_expenses.any(DatasourceExpense.updated_at < today)` ?